src/transformers/modeling_utils.py

```diff
@@ -2186,6 +2188,18 @@ def from_pretrained(cls, pretrained_model_name_or_path: Optional[Union[str, os.P
                 "The argument `trust_remote_code` is to be used with Auto classes. It has no effect here and is"
                 " ignored."
             )
+
+        # change device_map into a map if we passed an int, a str or a torch.device
+        if isinstance(device_map, torch.device):
+            device_map = {"": device_map.index}
```

Comment:
```suggestion
            device_map = {"": device_map}
```
No need to index, it will work like this.